framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -307,6 +307,19 @@ public Receiver visit(NameExpr expr, FlowExpressionContext context) {
             }
 
             if (fieldElem != null && fieldElem.getKind() == ElementKind.FIELD) {
+                if (!originalReceiver
+                        && TreeUtils.enclosingClass(path) != null
```

Comment:
`path` is the TreePath to the expression (in source code) whose type has an annotation with the FlowExpression.  So, this code won't work correctly.  I added a failing test case.